dask_kubernetes/common/networking.py

```diff
@@ -196,8 +196,14 @@ async def get_scheduler_address(
 async def wait_for_scheduler(api, cluster_name, namespace, timeout=None):
     pod_start_time = None
     while True:
+        async with kubernetes.client.api_client.ApiClient() as api_client:
+            k8s_api = kubernetes.client.CoreV1Api(api_client)
+            pods = await k8s_api.list_namespaced_pod(
+                namespace=namespace,
+                label_selector="dask.org/component=scheduler-deployment",
+            )
```

Comment:
We probably also need to add another selector for the `dask.org/cluster={cluster_name}` to make sure we are selecting just the scheduler we want.